arctic/serialization/numpy_arrays.py

```diff
@@ -203,13 +203,17 @@ def deserialize(self, data, columns=None):
         columns: None, or list of strings
             optionally you can deserialize a subset of the data in the SON. Index
             columns are ALWAYS deserialized, and should not be specified
+        inplace: Convert and remove items from data in-place
+            this will modify data
 
         Returns
         -------
         pandas dataframe or series
         """
         if not data:
             return pd.DataFrame()
+        if not inplace:
+            data = data[:]
```

Comment:
Does this hurt the case when there is no index?  e.g. you are making a defnesive copy even though you won't hit the set_index call?

Reply:
data is a list of the raw bytes. To save memory I allow the user to convert the list into dataframes in-place. This switch copies list first (shallow) so the caller doesn't get their reference to data changed. No impact to the index.